driver-core/src/main/com/mongodb/internal/connection/InternalConnectionInitializer.java

```diff
@@ -17,12 +17,11 @@
 package com.mongodb.internal.connection;
 
 import com.mongodb.internal.async.SingleResultCallback;
-import com.mongodb.connection.ConnectionDescription;
 
 interface InternalConnectionInitializer {
 
-    ConnectionDescription initialize(InternalConnection internalConnection);
+    InternalConnectionInitializationDescription initialize(InternalConnection internalConnection);
```

Comment:
Now we need both a ConnectionDescription and a ServerDescription, which are not quite the same thing